doc/userguide/src/CreatingTestData/CreatingTestCases.rst

```diff
@@ -904,6 +904,25 @@ all the looped elements even if there are failures.
            1st arg    ${index}
        END
 
+Templates with IF block
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+`if block`_ can be also used together with templates.
```

Comment:
I'm still not certain is "block" good or should it be e.g. "syntax". What's certain is that "if" is wrong case in the beginning of a sentence. I think "IF" would be good because it demonstrates the actual syntax as well ,l but normal "If" could be fine as well. Should check has "FOR" or "for" been used with for loops.